test/main/store/actions/index.test.js

```diff
@@ -984,6 +986,70 @@ describe('#setBlockHeight', () => {
   })
 })
 
+describe('#updateAccountMeta', () => {
+  const realDateNow = Date.now.bind(global.Date)
+  let main
+
+  const updaterFn = (node, accountMetaId, update) => {
+    expect(node).toBe('main.accountsMeta')
+    main.accountsMeta[accountMetaId] = update(main.accountsMeta[accountMetaId])
+  }
+
+  beforeEach(() => {
+    main = {
+      accountsMeta: {
+        1: {
+          name: 'cool account',
+          lastUpdated: 1568682918135
+        }
+      },
+    }
+    global.Date.now = jest.fn(() => new Date('2022-11-17T11:01:58.135Z').valueOf())
+  })
+
+  afterEach(() => {
+    global.Date.now = realDateNow
+  })
+
+  const setAccountMeta = (accountMetaId, name, lastSignerType) => updateAccountMetaAction(updaterFn, accountMetaId, { name, lastSignerType })
+
+  it('should create a new value with the expected data', () => {
+    setAccountMeta(2, 'not so cool account', 'seed')
+
+    expect(main.accountsMeta).toStrictEqual({ 
+      1: { name: 'cool account', lastUpdated: 1568682918135 }, 
+      2: { name: 'not so cool account', lastUpdated: 1668682918135 }
+    })
+  })
+
+  it('should update an existing value with the expected data', () => {
+    setAccountMeta(1, 'not so cool account', 'seed')
+
+    expect(main.accountsMeta).toStrictEqual({
+      1: { name: 'not so cool account', lastUpdated: 1668682918135 }
+    })
+  })
+
+  const accountTypes = ['ring', 'seed', 'ledger', 'trezor', 'lattice', 'aragon']
```

Comment:
I don't think we need all 6 of these tests, they're really all doing the same thing. also are we not allowing users to set an account name to "Trezor Account" if they wanted to?

Reply:
Also good to think about the important edge case of a user having access to someone's Frame instance that they want to prank, we need to support the ability to rename, e.g. all Ledger accounts to Trezor, etc. </sarcasm>

How do you propose to do this then?  Store a flag `isDefaultName` for each account that gets switched on the initial change of the name?

Reply:
Implementation stays the same as discussed, will update tests